intranet/apps/polls/models.py

```diff
@@ -107,14 +107,10 @@ def can_vote(self, user):
         if not self.in_time_range():
             return False
 
-        if self.groups.count() == 0:
+        if not self.groups.exists():
             return True
 
-        for g in self.groups.all():
-            if g in user.groups.all():
-                return True
-
-        return False
+        return user.groups.filter(id__in=self.groups.values_list("id", flat=True))
```

Comment:
You can use `intersect` here (https://docs.djangoproject.com/en/1.11/ref/models/querysets/#intersection)
```suggestion
        return user.groups.intersection(self.groups)
```